server/core/region.go

```diff
@@ -406,6 +412,30 @@ func (r *RegionInfo) GetStat() *pdpb.RegionStat {
 	}
 }
 
+// SetBuckets sets the buckets of the region.
+func (r *RegionInfo) SetBuckets(buckets *metapb.Buckets) bool {
```

Comment:
I prefer not to use `setxxx` for region directly.

Reply:
How about using UpdateBuckets?